### DISCUSSION TITLE
R32Uint StorageTexture not enabled by default?

### DISCUSSION THREAD

**Discussion by OneKnightOnly** [gfx-rs/wgpu#2734] [Q&A] [answered]
R32Uint StorageTexture is listed in both the wgsl and webgpu spec, so why does it require the feature TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES to be enabled?

**Comment by cwfitzgerald** [comment, marked as answer]
You shouldn't need it, it's listed as all_flags in the describe() function: https://github.com/gfx-rs/wgpu/blob/master/wgpu-types/src/lib.rs#L2150.

What's the exact error you're getting?

**Reply by OneKnightOnly** [reply]
Error: 
In Device::create_bind_group_layout
 binding 0 entry is invalid
 Features TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES are required but not enabled on the device
    
Thats with features set to Features::empty(),
But if I use Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES,
then it compiles without an error.

Is this a bug?

**Reply by cwfitzgerald** [reply]
What backend is that using? Either way, yes it is a bug.

**Reply by OneKnightOnly** [reply]
Thats with the vulkan backend, on windows 10.

**Reply by cwfitzgerald** [reply]
Yup, definitely a bug

**Reply by OneKnightOnly** [reply]
I created a bug report
